text/0007-ownership-custodians.md

```diff
@@ -0,0 +1,113 @@
+
+I propose we introduce a new clearly documented and audited process to deal with these cases. Some members of the RubyGems community are nominated as custodians (could include members of Ruby core team for example, admins on RubyGems, etc). Some pre-requisites like experience, activity, etc may be considered. Those members have the ability to go through a gem adoption process either for themselves or on behalf of another member of the community.
+
+## Guide-level explanation
+
+### Gem Adoption Process
+
+- Given that a user makes an ownership request on a specific gem, after a certain grace period (e.g. 7 days) they can request a gem custodian review.
+- The gem custodian can review all details of the user making the request, including all other gem ownership requests they have made, and the details of the current gem ownership request.
+- Within certain guidelines, they have the ability to approve a gem ownership request without the express consent of the current owner.
+	- The gem must meet the general criteria of being abandoned.
+- The gem custodian can attempt to communicate with the current owner on behalf of the ownership request.
+	- The current owner can contest the ownership transfer but this is not a veto power.
+	- The gem custodian is advised to give sufficient time, at least 30 days, for the current owner to respond.
+- Based on this investigation, the gem custodian can approve the ownership request.
+- Requests of this nature should be peer-reviewed, and at least one gem custodian independent of the user making the ownership request should confirm the ownership request - this means that gem custodians cannot approve their own requests.
+	- Gems which do not meet the requirements for being abandoned require the approval of 3 gem custodians.
+	- Any gem custodian can veto a gem ownership request.
+
+If this process fails because the current owner rejected the transfer, the ownership request owner can re-request a gem custodian after a period of 12 months. The gem custodian should consider this history in their decision making process.
+
+### Abandoned Gem Criteria
+
+A gem is considered "in use" if the gem itself, or any of its direct reverse dependencies (version specific) have more than 10,000 downloads in the past year.
+
+A gem is considered "maintained" if the author has logged into <https://RubyGems.org> within the past 12 months.
+
+A gem is considered to be "valid" if the homepage and other related metadata is valid (e.g. URLs return relevant pages).
+
+A gem is considered to be "working" if a user can check out and run the unit tests on a supported, non-EOL Ruby implementation.
+
+A gem is considered to be "stable" if it has a published release `>= 1.0.0`.
```

Comment:
These are great ideal criteria. Unfortunately, the reality is a bit nuanced. I would recommend removing following criteria:  
>  the homepage and other related metadata is valid

92% of all version have empty metadata.

>  non-EOL Ruby implementation.

more than 80% of traffic we receive is from ruby version which have reached EOL https://ecosystem.rubytogether.org/

>  it has a published release `>= 1.0.0`.

rubocop didn't have a stable release for over 8 years.

IMHO, we should stick to activity-related metrics like owner responsiveness and recent releases. Downloads can be a good measure if we filter the noise. Further as of now rubygems.org only tracks total downloads (not downloads over a period).

Reply:
@sonalkr132 we could also map out ownership and owners activities as a factor. I do it already for internal data mining. When owners are active, the change that their packages are not "abandoned" per se is higher. This coulda act as one of the signals.